src/lib/class/Client/core.cpp

```diff
@@ -0,0 +1,109 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   core.cpp                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: mcourtoi <mcourtoi@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/02/20 21:20:40 by mcourtoi          #+#    #+#             */
+/*   Updated: 2024/02/28 17:31:48 by mcourtoi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "class/Client.hpp"
+
+Client::Client(
+	int const       socket,
+	Nickname const &nickname,
+	Hostname const &hostname,
+	Username const &username,
+	Realname const &realname,
+	UserModes const modes) :
+	_socket(socket),
+	_msg_in(),
+	_msg_out(),
+	_nickname(nickname),
+	_hostname(hostname),
+	_username(username),
+	_realname(realname),
+	_modes(modes),
+	_time_last_msg(std::clock())
```

Comment:
Yes because I first need to create a separate epoll_socket with a function epoll_create() and that can fail so I prefer to initialize it after that's done.